lib/utils.js

```diff
@@ -116,11 +117,26 @@ export function mkdir(dirName: string): Promise<void> {
   })
 }
 
-export function rmdir(dirName: string): Promise<void> {
-  return new Promise((resolve, reject) => {
-    // The keybase service will have made some files here; we want to clean them all out
-    // Node doesn't have a recursive dir clean, but.
-    console.log(`TODO: implement recurseive rmdir for tmp directory ${dirName}`)
-    resolve()
-  })
+export async function rmdirRecursive(dirName: string): Promise<void> {
+  const fsLstat = promisify(fs.lstat)
+  const fsUnlink = promisify(fs.unlink)
+  const fsRmdir = promisify(fs.rmdir)
+  const fsReaddir = promisify(fs.readdir)
+  try {
+    const dirStat = await fsLstat(dirName)
+    if (dirStat) {
+      for (const entry of await fsReaddir(dirName)) {
+        const entryPath = path.join(dirName, entry)
+        const stat = await fsLstat(entryPath)
+        if (stat.isDirectory()) {
+          await rmdirRecursive(entryPath)
+        } else {
+          await fsUnlink(entryPath)
+        }
+      }
+      await fsRmdir(dirName)
+    }
+  } catch (error) {
+    return error
```

Comment:
I'm not sure we want to return the error here. We should either:

1. Let the error propagate up and handle it in `deinit`.
2. Handle the error, probably via logging to `console.error`.

I think I'm more in favor of handling it in `deinit` and logging that we couldn't delete it directory in there as that keeps `rmdirRecursive` a little more generic in case we need to use it in a future part of the bot and handle errors differently.

Reply:
and ah right, I was basically doing this wrong by returning an error, that would've effectively made the promise resolve instead of reject, but with an error? is that right?

Reply:
yup! The promise would return the error when it resolves and never reject. This also doesn't really make sense because our function signature says we're returning `void`.